runners/flink/runner/src/main/java/org/apache/beam/runners/flink/FlinkRunnerResult.java

```diff
@@ -84,6 +87,6 @@ public State waitUntilFinish() {
 
   @Override
   public State waitUntilFinish(Duration duration) {
-    throw new UnsupportedOperationException("FlinkRunnerResult does not support waitUntilFinish.");
+    return State.DONE;
```

Comment:
Could you comment that waitUntilFinish and getState is no-op because FlinkRunner does blocking execution, and add link to https://issues.apache.org/jira/browse/BEAM-593